HIP/hip-1340.md

```diff
@@ -0,0 +1,388 @@
+
+Delegation information will be represented in a form of a Delegation Indicator - a fixed prefix of `0xef0100` followed by 20-byte EVM address of the delegation target.
+
+```
+delegation_indicator = 0xef0100 || 20-byte address
+```
+
+The Delegation Indicator will be stored as the code of an EOA using the facilities that currently exist for storing the code (runtime bytecode) of contract accounts - i.e. the `ContractStateStore` interface. To be able to utilize the interface, EOA's AccountID will be directly converted to a corresponding ContractID.
+
+The main revision is that the code of an EOA will become mutable. It can be either empty or contain a Delegation Indicator. No other code is allowed.
+
+As a result, all code-related queries previously available for contract accounts should now also function for EOAs, which include:
+- `ContractGetBytecodeQuery`
+- `eth_getCode` RPC method
+
+Since the code of a regular contract cannot begin with `0xef` (as per EIP-3541), only the code of an EOA can contain a Delegation Indicator.
+
+The delegation target address is an arbitrary 20-byte address. No additional validation is done at the time of storing the Delegation Indicator (e.g. whether it points to a valid contract, etc), the only exception being `0x0000000000000000000000000000000000000000` - any attempt to store it will result in clearing the Delegation Indicator instead (i.e. setting the bytecode to an empty byte array).
+
+#### Updating Code Delegation of an EOA
+
+We propose to enable two ways of setting/updating the Code Delegation. One using Hedera-specific API, and another one using a new EVM transaction type following the specification of EIP-7702.
+
+##### Updating Code Delegation Via CryptoCreate or CryptoUpdate Transaction
+
+We propose to extend `CryptoCreateTransactionBody` and `CryptoUpdateTransactionBody` with an additional field holding a 20-byte EVM address that the user wishes to setup as a delegation target for their account:
+
+```
+message CryptoCreateTransactionBody {
+...
+
+  /**
+  * The delegated contract address for the account.
+  * If this field is set, a call to the account's address within a smart contract will
+  * result in the code of the authorized contract being executed.
+  */
+  bytes delegation_address = 20;
+}
+
+message CryptoUpdateTransactionBody {
+...
+
+  /**
+  * The delegated contract address for the account.
+  * If this field is set, a call to the account's address within a smart contract will
+  * result in the code of the authorized contract being executed.
+  */
+  bytes delegation_address = 21;
+}
+```
+
+If set, the value of the field will be checked to ensure it's a valid 20-byte EVM address. It will then be prefixed with the Delegation Indicator prefix, and stored as the Delegation Indicator for the given account. No other validation will be performed at the time of setting the Delegation Indicator (e.g. no validation that the target address exists, isn't a precompile, etc).
```

Comment:
Is this how Ethereum does it also? Could you delegate to a system contract?
I believe one was added in Cancun right or maybe it was pectra, there was a contract added that was EOA submitted that provided a few methods. 

I'm just curious why we wouldn't just check here and stop to avoid having to check later during the EOA pathway execution.
If checking here though would result in incompatibility or undesired user flow that only happen on Hiero then it makes sense not to check. We should not that though